core/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceParser.java

```diff
@@ -36,45 +34,16 @@
 /**
  *
  */
-public abstract class AbstractValuesSourceParser<VS extends ValuesSource>
-        implements Aggregator.Parser {
+public abstract class ValuesSourceParser<VS extends ValuesSource> implements Aggregator.Parser {
     static final ParseField TIME_ZONE = new ParseField("time_zone");
 
-    public abstract static class AnyValuesSourceParser extends AbstractValuesSourceParser<ValuesSource> {
-
-        protected AnyValuesSourceParser(boolean scriptable, boolean formattable) {
-            super(scriptable, formattable, false, ValuesSourceType.ANY, null);
-        }
-    }
-
-    public abstract static class NumericValuesSourceParser extends AbstractValuesSourceParser<ValuesSource.Numeric> {
+    protected boolean scriptable = true;
```

Comment:
Probably for another PR since it's unrelated but I wonder if `scriptable`, `formattable` and ` timezoneAware` should be properties of the Builder object rather than/ as well as the parser so the builder can ensure an IllegalArgumentException is thrown if e.g. an unscriptable agg has the script method called on it? /cc @jpountz